zkevm/src/utils.rs

```diff
@@ -0,0 +1,82 @@
+use crate::circuit::DEGREE;
+use halo2_proofs::arithmetic::BaseExt;
+use halo2_proofs::poly::commitment::Params;
+use pairing::bn256::{Bn256, Fr, G1Affine};
+use rand::SeedableRng;
+use rand_xorshift::XorShiftRng;
+use std::fs::File;
+use std::io::{BufReader, Read, Result, Write};
+use std::path::Path;
+use zkevm_circuits::evm_circuit::param::STEP_HEIGHT;
+use zkevm_circuits::evm_circuit::witness::Block;
+
+/// generate randomness for the block
+pub fn load_randomness(block: Block<Fr>) -> Vec<Box<[Fr]>> {
+    let power_of_randomness: Vec<Box<[Fr]>> = (1..32)
+        .map(|exp| {
+            vec![
+                block.randomness.pow(&[exp, 0, 0, 0]);
+                block.txs.iter().map(|tx| tx.steps.len()).sum::<usize>() * STEP_HEIGHT
+            ]
+            .into_boxed_slice()
+        })
+        .collect();
+    power_of_randomness
+}
+
+/// return setup params by reading from file or generate new one
+pub fn load_or_create_params(params_path: &str) -> Result<Params<G1Affine>> {
+    if Path::new(params_path).exists() {
+        load_params(params_path)
+    } else {
+        create_params(params_path)
+    }
+}
+
+pub fn load_params(params_path: &str) -> Result<Params<G1Affine>> {
+    log::info!("start load params");
+    let f = File::open(params_path)?;
+    let p = Params::read::<_>(&mut BufReader::new(f))?;
+    log::info!("load params successfully!");
+    Ok(p)
+}
+
+pub fn create_params(params_path: &str) -> Result<Params<G1Affine>> {
+    log::info!("start create params");
+    let params: Params<G1Affine> = Params::<G1Affine>::unsafe_setup::<Bn256>(DEGREE as u32);
+    let mut params_buf = Vec::new();
+    params.write(&mut params_buf)?;
+
+    let mut params_file = File::create(&params_path)?;
+    params_file.write_all(&params_buf[..])?;
+    log::info!("create params successfully!");
+    Ok(params)
+}
+
+/// return setup rng by reading from file or generate new one
+pub fn load_or_create_rng(seed_path: &str) -> Result<XorShiftRng> {
+    if Path::new(seed_path).exists() {
+        load_rng(seed_path)
+    } else {
+        create_rng(seed_path)
+    }
+}
```

Comment:
```suggestion
/// return setup seed by reading from file or generate new one
pub fn load_or_create_seed(seed_path: &str) -> Result<SEED_TYPE_XXXXXX> {
    if Path::new(seed_path).exists() {
        load_seed(seed_path)
    } else {
        create_seed(seed_path)
    }
}
```

Reply:
Why not use `rng` instead of `seed`? We should let the callers use seed generate rng themselves?